metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -389,6 +408,7 @@ def query(query):
             SourceCapability.LINEAGE_COARSE: "Current role does not have permissions to snowflake account usage views",
             SourceCapability.LINEAGE_FINE: "Current role does not have permissions to snowflake account usage views",
             SourceCapability.USAGE_STATS: "Current role does not have permissions to snowflake account usage views",
+            SourceCapability.TAGS: "Either no tags have been applied to objects or the current role does not have permission to access the objects",
```

Comment:
Do we need to have any specific access privileges to be able to list and view tags on objects ?

Reply:
Per the [documentation](https://docs.snowflake.com/en/user-guide/object-tagging.html#summary-of-ddl-commands-operations-and-privileges):

- the `tag_references` and `tag_references_all_columns`-functions: "Results are only returned for a role that has access to the specified object.". [docs1](https://docs.snowflake.com/en/sql-reference/functions/tag_references.html#usage-notes), [docs2](https://docs.snowflake.com/en/sql-reference/functions/tag_references_all_columns.html). It's my understanding that the role doesn't need explicit access to the information schema, only general USAGE on the database. 
- `snowflake.account_usage.tag_references` is of course part of the account_usage schema, which users are already instructed to grant usage to.

Reply:
Now that I checked more closely, I noticed the note about the account usage schema in the docs. I'll add tags as one possible use case for that privilege.

Reply:
We will also need to update the `check_capabilities` method for "Test Connection" feature to mention whether extracting tags is supported for configured role.

Reply:
Done